### PR TITLE
contracts: feature: introduce realm levels

### PR DESCRIPTION
**Pull request by credence0x**
contracts for #1719
- introduce realm levels
- create a function for upgrading realm levels as well as functions for configuration
- restrict building on certain hexes depending on realm level
- tests

### CODE REVIEW

File: contracts/src/models/realm.cairo [hunk 1]
```diff
@@ -21,15 +21,16 @@ pub struct Realm {
...
 
-trait RealmCustomTrait {
-    fn has_resource(self: Realm, resource_type: u8) -> bool;
-    fn assert_is_set(self: Realm);
-}
-
+#[generate_trait]
 impl RealmCustomImpl of RealmCustomTrait {
+    fn max_level(self: Realm) -> u8 {
+        3
```
Comment (ponderingdemocritus) [inline contracts/src/models/realm.cairo:31]: change this to config

**Review by ponderingdemocritus — APPROVED** [review]
smol config change

**Comment by ponderingdemocritus** [comment]
In config i have added some contants which can be used to init the config, can you update to use them

File: contracts/src/models/realm.cairo (hunk 1 above)
Comment (credence0x) [reply contracts/src/models/realm.cairo]: updated

**Pull request merged**